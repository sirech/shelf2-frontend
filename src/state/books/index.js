import * as constants from './constants'
import * as actions from './actions'

export { actions, constants }
export { default as reducer } from './reducers'
