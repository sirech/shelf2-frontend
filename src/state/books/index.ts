import * as constants from './constants'
import * as actions from './slice'

export { actions, constants }
export { default as booksSelector } from './selectors'
export { default as reducer } from './slice'
