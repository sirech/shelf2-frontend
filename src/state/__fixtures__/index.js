import R from 'ramda'

import books from './books'
import years from './years'
import form from './form'
import login from './login'

const modal = () => ({
  opened: false
})

const search = R.pipe(books, R.omit(['activeYear']))

const state = () => {
  return {
    books: books(),
    search: search(),
    years: years(),
    form: form(),
    modal: modal(),
    login: login()
  }
}

export { books, search, years, form, login }
export default state
