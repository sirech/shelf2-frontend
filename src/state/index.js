import { combineReducers } from 'redux'

import * as R from 'ramda'

import { constants as booksConstants, reducer as booksReducer } from './books'
import {
  constants as searchConstants,
  reducer as searchReducer,
} from './search'
import { constants as yearsConstants, reducer as yearsReducer } from './years'
import { constants as formConstants, reducer as formReducer } from './form'
import { constants as modalConstants, reducer as modalReducer } from './modal'
import { constants as loginConstants, reducer as loginReducer } from './login'

export const actionPicker = list => {
  return actions => {
    const h = R.pick(list)(actions)

    if (Object.keys(h).length !== list.length) {
      throw new Error(
        `Not all requested actions (${list}) exist in the given actions`
      )
    }

    return h
  }
}

const rootReducer = combineReducers({
  [booksConstants.namespace]: booksReducer,
  [searchConstants.namespace]: searchReducer,
  [yearsConstants.namespace]: yearsReducer,
  [formConstants.namespace]: formReducer,
  [modalConstants.namespace]: modalReducer,
  [loginConstants.namespace]: loginReducer,
})

export default rootReducer
