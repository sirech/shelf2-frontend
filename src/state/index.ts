import { ThunkAction, UnknownAction, configureStore } from '@reduxjs/toolkit'

import { constants as booksConstants, reducer as booksReducer } from './books'
import {
  constants as searchConstants,
  reducer as searchReducer,
} from './search'
import { constants as yearsConstants, reducer as yearsReducer } from './years'
import { constants as loginConstants, reducer as loginReducer } from './login'

const store = configureStore({
  reducer: {
    [booksConstants.namespace]: booksReducer,
    [searchConstants.namespace]: searchReducer,
    [yearsConstants.namespace]: yearsReducer,
    [loginConstants.namespace]: loginReducer,
  },
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  UnknownAction
>

export default store
