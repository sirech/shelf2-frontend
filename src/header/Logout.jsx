// @flow

import React from 'react'
import { useDispatch } from 'react-redux'

import { NavItem } from 'reactstrap'

import { actions } from 'state/login'

const Logout = () => {
  const dispatch = useDispatch()

  return (
    <NavItem>
      <div
        onClick={() => dispatch(actions.logout())}
        className="nav-link"
        role="link"
      >
        Logout
      </div>
    </NavItem>
  )
}

export default Logout
