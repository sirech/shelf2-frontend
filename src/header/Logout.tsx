import React from 'react'

import { NavItem } from 'reactstrap'
import { useAppDispatch } from 'hooks'

import { actions } from 'state/login'

const Logout = () => {
  const dispatch = useAppDispatch()

  return (
    <NavItem
      onClick={() => dispatch(actions.logout())}
      className="nav-link"
      role="link"
    >
      Logout
    </NavItem>
  )
}

export default Logout
