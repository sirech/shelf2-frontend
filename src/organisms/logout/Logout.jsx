// @flow

import React from 'react'
import { connect } from 'react-redux'

import R from 'ramda'
import { NavItem } from 'reactstrap'

import authenticated from '../../organisms/authenticated'
import { actions } from '../../state/login'

type Props = {
  logout: void => void
}

class Logout extends React.Component {
  props: Props
  onClick: Function

  constructor () {
    super()
    this.onClick = this.onClick.bind(this)
  }

  onClick () {
    this.props.logout()
  }

  render () {
    return (
      <NavItem>
        <a href='#' onClick={this.onClick} className='nav-link'>Logout</a>
      </NavItem>
    )
  }
}

export default authenticated()(connect(
  null,
  R.pick(['logout'])(actions)
)(Logout))
