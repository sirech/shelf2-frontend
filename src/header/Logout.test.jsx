import React from 'react'

import Logout from './Logout'
import { fullRender } from 'test'
import state from 'state/__fixtures__'

describe('components', () => {
  describe('Logout', () => {
    it('renders correctly', () => {
      const { component } = fullRender(<Logout />, state())
      expect(component.toJSON()).toMatchSnapshot()
    })
  })
})
