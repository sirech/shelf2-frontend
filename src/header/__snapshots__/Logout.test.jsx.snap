// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Logout renders correctly 1`] = `
<li
  className="nav-link nav-item"
  onClick={[Function]}
  role="link"
>
  Logout
</li>
`;
